src/signal_generator.cpp

```diff
@@ -71,6 +71,9 @@
 
 #define AMPLITUDE_VOLTS	5.0
 
+#define MAX_PREVIEW_RANGE (float)(SHRT_MAX)
```

Comment:
Avoid using macros for constants, you could use instead:
`static const float MAX_PREVIEW_RANGE`